libsignal-service-actix/Cargo.toml

```diff
@@ -12,7 +12,9 @@ libsignal-protocol = { git = "https://github.com/Michael-F-Bryan/libsignal-proto
 
 awc = { version = "2.0.0", features=["rustls"] }
 actix = "0.10.0"
+actix-http = "2.1.0"
 actix-rt = "1.1"
+actix-multipart-rfc7578 = {version = "0.3.0-rc", git="https://github.com/rubdos/rust-multipart-rfc7578.git", branch="actix-2"} # Not an "official" actix project
```

Comment:
Tracking PR https://github.com/ferristseng/rust-multipart-rfc7578/pull/19

Reply:
@gferon this crate does seem kind-of abandoned. Would you have any suggestion how to resolve this? Maybe contribute it to Actix directly?

Reply:
Actix is tracking that here: https://github.com/actix/actix-web/issues/195

So I guess I'll take a crack at that to get this merged.

Reply:
That issue is quite some work... I gave it a shot, then noticed that one of the Actix devs has the `mpart-async` crate that solves this a lot cleaner. I've rewritten it with that, now I'll clean that up a bit.

Reply:
Some extra comments:

- Actix basically expects `Stream<Item=Bytes>` for its chunks
- I think `tokio::io::AsyncRead` would be the "logical" block
- Converting `tokio::io::AsyncRead` into `Stream<Item=Bytes>` is [non-trivial](https://github.com/rust-lang/futures-rs/issues/2006).
- The current `std::io::Read` is *not good*. It blocks. That's terrible. We're allocating, that's slightly less terrible, but still bad.

I'd like to merge this this weekend, @gferon, you got any suggestions how to clean this up? Just pass on the whole `Vec<u8>` for the `cdn0` implementation, and when we start #34, do it decently (because that's an actual streaming API)? The `cdn0` version needs to know `Content-Length` anyway, so we might as well keep this kinda dirty.

Reply:
> Just pass on the whole `Vec<u8>` for the `cdn0` implementation, and when we start #34, do it decently (because that's an actual streaming API)? The `cdn0` version needs to know `Content-Length` anyway, so we might as well keep this kinda dirty.

This definitely sounds sensible, either `Vec<u8>` or `&[u8]` if the caller should retain ownership of the data somehow.